pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -592,3 +594,23 @@ func (r *ReconcileLogStorage) getKibanaService(ctx context.Context) (*corev1.Ser
 	}
 	return &svc, nil
 }
+
+func calculateFlowShards(nodesSpecifications *operatorv1.Nodes, defaultShards int) int {
+	if nodesSpecifications == nil || nodesSpecifications.ResourceRequirements == nil || nodesSpecifications.ResourceRequirements.Requests == nil {
+		return defaultShards
+	}
+
+	var nodes = nodesSpecifications.Count
+	var cores, _ = nodesSpecifications.ResourceRequirements.Requests.Cpu().AsInt64()
+	var shardPerNode = int(cores) / 4
+
+	if nodes <= 0 || cores <= 0 || shardPerNode <= 0{
+		return defaultShards
+	}
+
+	if shardPerNode < defaultShards {
```

Comment:
I don't think with the new default of 1 that this case will ever be used, since shardPerNode is calculated with integer math.